Packages/io.chainsafe.web3-unity/Runtime/Scripts/UI/ConnectionProviderButton.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace ChainSafe.Gaming.UnityPackage.UI
+{
+    public class ConnectionProviderButton : MonoBehaviour
+    {
+        public Image Icon;
```

Comment:
why are these serialized fields public? we can make all of these private

Reply:
If you can help it access modifiers should be private to avoid referencing important code bits to enforce better design I guess, say someone references the button from another script and adds a listener to it, ideally we want the button to be referenced in that script for easy access and readability...that kinda stuff, it's also private by default xD

Reply:
I think it's also one of the clean code principles